test/test-fs-readdir.c

```diff
@@ -460,3 +461,88 @@ TEST_IMPL(fs_readdir_non_empty_dir) {
   MAKE_VALGRIND_HAPPY(uv_default_loop());
   return 0;
  }
+
+static void readdir_symlink_readdir_cb(uv_fs_t* req) {
+  uv_dir_t* dir;
+
+  ASSERT_PTR_EQ(req, &readdir_req);
+  ASSERT_EQ(req->fs_type, UV_FS_READDIR);
+  dir = req->ptr;
+
+  if (req->result == 0) {
+    uv_fs_req_cleanup(req);
+    ASSERT_EQ(3, non_empty_readdir_cb_count);
+    uv_fs_closedir(uv_default_loop(),
+                   &closedir_req,
+                   dir,
+                   non_empty_closedir_cb);
+  } else {
+    if (strcmp(symlink_dirents[0].name, "test_symlink") == 0) {
```

Comment:
Why the check, if we are asserting the same?

Reply:
There are 2 folders named `test_symlink` and `test_subdir` in `symlink_dirents`. I just want to check the type of `test_symlink`, so I added this `if` condition.

Reply:
Ok.